src/components/document-viewer/doc-fragment-creator.jsx

```diff
@@ -16,59 +18,106 @@ class DocFragmentCreator extends Component {
+      lintError: null,
+      validationError: null,
+      message: '',
+      sourceCode,
     };
 
     this.saveChange = this.saveChange.bind(this);
   }
 
+  handleChange() {
+    const sourceCode = this.refs.ace.editor.getValue();
+    this.setState({ sourceCode });
+    lint(sourceCode)
+      .then(() => {
+        this.setState({ lintError: null });
+      })
+      .catch((err) => {
+        this.setState({ lintError: err.message });
+      });
+  }
+
   saveChange() {
     const source = this.refs.ace.editor.getValue();
-    DocumentService.addFragment(source);
+    const message = `Added section using Pensieve\n\n${this.state.message}`;
 
-    this.setState({
-      loading: true,
-    });
-    GitHubService.commit({ content: DocumentService.getSource() }).then(() => {
-      this.setState({
-        loading: false,
+    lint(source)
+      // When validating, strip the title field from the source
+      .then(() => schemaValidate(this.props.schema, source.replace(/^.+\n/, '')))
+      .then(() => {
+        DocumentService.addFragment(source);
+        this.setState({
+          loading: true,
+          validationError: null,
+        });
+        GitHubService.commit({ content: DocumentService.getSource(), message }).then(() => {
+          this.setState({
+            loading: false,
+          });
+          this.props.close();
+        });
+      })
+      .catch(PensieveLinterError, (err) => {
+        this.setState({ lintError: err.message });
+      })
+      .catch(PensieveValidationError, (err) => {
+        this.setState({ validationError: err.message });
       });
-      this.props.close();
-    });
   }
 
   render() {
-    const json = {};
-    _.forEach(this.props.schema, (value, key) => {
-      const defaultFieldValue = `// ${value.type} value`;
-      json[key] = defaultFieldValue;
-    });
-    const entryName = `New Entry - ${new Date().toString()}`;
-    const entry = {};
-    entry[entryName] = json;
-    const sourceCode = jsyaml.safeDump(entry);
-    console.log(sourceCode);
     return (
       <div>
-        {this.state.loading
-          ? <FontAwesome spin name="cog" />
-          : <Button onClick={this.saveChange}>Save</Button>}
-
         <AceEditor
           mode="yaml"
           theme="chrome"
           name="code"
           width="100%"
           ref="ace"
           fontSize={14}
-          onChange={this.onChange}
-          value={sourceCode}
+          onChange={() => this.handleChange()}
+          value={this.state.sourceCode}
           onLoad={(editor) => {
             editor.focus();
             editor.getSession().setUseWrapMode(true);
           }}
         />
+        {this.state.lintError &&
+          <Text color="red">
+            {this.state.lintError}
+          </Text>}
+        {this.state.validationError &&
+          <Text color="red">
+            {this.state.validationError}
+          </Text>}
+        {this.state.loading
+          ? <FontAwesome spin name="cog" />
```

Comment:
I was thinking about our discussion on conditional rendering the other day, and thought it'd be cool to make a higher order component where you can pass a condition for conditional rendering, ofc there are already those libs [already exist](
https://www.npmjs.com/package/conditional-react-component), still not sure which syntax I prefer but want to try it one.